finetuner/utils.py

```diff
@@ -0,0 +1,115 @@
+import csv
+from contextlib import nullcontext
+from typing import Generator, Optional, TextIO, Union
+
+from docarray import Document
+from docarray.document.generators import _subsample
+from docarray.document.mixins.helper import _is_uri
+
+from finetuner.constants import __DEFAULT_TAG_KEY__
+
+
+def from_csv(
+    file: Union[str, TextIO],
+    task: str = 'text-to-text',
+    size: Optional[int] = None,
+    sampling_rate: Optional[float] = None,
+    dialect: Union[str, 'csv.Dialect'] = 'auto',
+    encoding: str = 'utf-8',
+    is_labeled: bool = False,
+) -> Generator['Document', None, None]:
+    """
+    Takes a csv file and returns a generator of documents, with each document containing
+    the information form one line of the csv
+
+    :param file: Either a filepath to or a stream of a csv file.
+    :param task: Specifies the modalities of the model that the returned data is to
+        be used for. This data is retrieved using the model name, and does not need
+        to be added to the csv_options argument when calling :meth:`finetuner.fit`
+    :param size: The number of rows to sample at once, 1 if left as None.
+    :param sampling_rate: The sampling rate between [0, 1].
+    :param dialect: A description of the expected format of the csv, can either be an
+        object of the :class:`csv.Dialect` class, or one of the strings returned by the
+        :meth:`csv.list_dialects()' function.
+    :param encoding: The encoding of the csv
+    :param is_labeled: Wether the second column of the csv represents a label that
+        should be assigned to the item in the first column (True), or if it is another
+        item that should be semantically close to the first.
+    :return: A generator of :class:`Document`s. Each document represents one line of the
+        imput csv.
+
+    """
```

Comment:
If the PR in https://github.com/jina-ai/finetuner-core/pull/372 is merged, we can add a argument to convert image uris to blobs. otherwise I would at least support to convert uris to image tensors since it does not make sense to upload dataset with uris pointing to local files